From 84477867e122251b8c47db240a8d699b0836aa6f Mon Sep 17 00:00:00 2001
From: Thorsten Blum <thorsten.blum@linux.dev>
Date: Thu, 30 Apr 2026 11:02:54 +0200
Subject: [PATCH 00010/15333] tty: xtensa/iss: use strnlen to improve
 iss_console_write

Use strnlen() to limit scanning 's' to 'count' bytes. Use the length of
's' to decide if simc_write() should be called instead of dereferencing
it first and then calling strlen().

With strnlen(), iss_console_write() is further hardened against callers
where 's' is not NUL-terminated.

Signed-off-by: Thorsten Blum <thorsten.blum@linux.dev>
Message-ID: <20260430090253.237654-3-thorsten.blum@linux.dev>
Signed-off-by: Max Filippov <jcmvbkbc@gmail.com>
---
 arch/xtensa/platforms/iss/console.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/xtensa/platforms/iss/console.c b/arch/xtensa/platforms/iss/console.c
index 8b95221375a8..8e54625cb2ba 100644
--- a/arch/xtensa/platforms/iss/console.c
+++ b/arch/xtensa/platforms/iss/console.c
@@ -166,8 +166,9 @@ late_initcall(rs_init);
 
 static void iss_console_write(struct console *co, const char *s, unsigned count)
 {
-	if (s && *s != 0)
-		simc_write(1, s, min(count, strlen(s)));
+	count = s ? strnlen(s, count) : 0;
+	if (count)
+		simc_write(1, s, count);
 }
 
 static struct tty_driver* iss_console_device(struct console *c, int *index)
-- 
2.55.0

